generators/templates.go

```diff
@@ -87,14 +87,70 @@ package client
 
 import (
 	"bytes"
+	"crypto/tls"
 	"encoding/json"
 	"errors"
 	"io/ioutil"
+	"os"
+	"os/user"
 	"net/http"
 
 	log "github.com/Sirupsen/logrus"
 )
 
+var httpClient *http.Client
+
+func init() {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+
+	httpClient = &http.Client{Transport: tr}
+}
+
+type NetctlConfig struct {
+	Token string ` + "`" + `json:"token"` + "`" + `
+}
+
+func addAuthTokenHeader(req *http.Request) error {
+
+	// if the request is not secure, we won't add the X-Auth-Token header
+	if req.URL.Scheme != "https" {
+		return nil
+	}
+
+	// construct path to user's netctl config file
+	usr, err := user.Current()
+	if err != nil {
+		return err
+	}
+
+	netctlConfigFile := usr.HomeDir + "/.netctl/config.json"
```

Comment:
So, instead of having netctl specific implict file, maybe AuthTokenHeader should just add in a generic token variable value, and netctl as client would manage that token variable by itself (thaw it to file, read it from file)